src/EnergyPlus/HWBaseboardRadiator.cc

```diff
@@ -1440,39 +1318,34 @@ namespace HWBaseboardRadiator {
         Real64 AA;
         Real64 BB;
         Real64 CC;
-        Real64 QZnReq;
         Real64 Cp;
-        auto &HWBaseboard = state.dataHWBaseboardRad->HWBaseboard;
-        auto &HWBaseboardDesignObject = state.dataHWBaseboardRad->HWBaseboardDesignObject;
-        auto &QBBRadSource = state.dataHWBaseboardRad->QBBRadSource;
-        auto &ZeroSourceSumHATsurf = state.dataHWBaseboardRad->ZeroSourceSumHATsurf;
-
-        ZoneNum = HWBaseboard(BaseboardNum).ZonePtr;
-        QZnReq = state.dataZoneEnergyDemand->ZoneSysEnergyDemand(ZoneNum).RemainingOutputReqToHeatSP;
-        AirInletTemp = HWBaseboard(BaseboardNum).AirInletTemp;
-        AirOutletTemp = AirInletTemp;
-        WaterInletTemp = HWBaseboard(BaseboardNum).WaterInletTemp;
-        WaterOutletTemp = WaterInletTemp;
-        WaterMassFlowRate = state.dataLoopNodes->Node(HWBaseboard(BaseboardNum).WaterInletNode).MassFlowRate;
+        auto &hWBaseboard = state.dataHWBaseboardRad->HWBaseboard(BaseboardNum);
+
+        int const ZoneNum = hWBaseboard.ZonePtr;
+        auto const &zeroSourceSumHATsurf = state.dataHWBaseboardRad->ZeroSourceSumHATsurf(ZoneNum);
+        Real64 QZnReq = state.dataZoneEnergyDemand->ZoneSysEnergyDemand(ZoneNum).RemainingOutputReqToHeatSP;
+        Real64 AirInletTemp = hWBaseboard.AirInletTemp;
+        Real64 WaterInletTemp = hWBaseboard.WaterInletTemp;
+        Real64 WaterMassFlowRate = state.dataLoopNodes->Node(hWBaseboard.WaterInletNode).MassFlowRate;
         HWBaseboardDesignData HWBaseboardDesignDataObject{
```

Comment:
You are creating a local copy of this object. Is that the intent? If not, change to `const &`.